From f40bec7a15873acd749a35c968d63db201a29e5f Mon Sep 17 00:00:00 2001
From: Fred Chen <Fred.Chen@quantatw.com>
Date: Thu, 23 Jul 2026 20:03:45 +0800
Subject: [PATCH] PSUSensor: add ADI MAX20912 and MAX20916 support
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add device ADI MAX20912 and MAX20916 reading support

The driver is under upstream review[1].
ADI commented that they’ll release their own version soon, so they
preferred that we hold off on submitting patch for now and wait for
theirs.
Setting to WIP until upstream kernel supports this driver.

[1] https://lore.kernel.org/all/20260707122701.751878-3-fredchen.openbmc@gmail.com/
gerrit: https://gerrit.openbmc.org/c/openbmc/dbus-sensors/+/92662

Change-Id: I697fddc7af123ed9236f44c8b89fee3461637b88
Signed-off-by: Fred Chen <Fred.Chen@quantatw.com>
---
 src/psu/PSUSensorMain.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/psu/PSUSensorMain.cpp b/src/psu/PSUSensorMain.cpp
index fd293a1..ba866f7 100644
--- a/src/psu/PSUSensorMain.cpp
+++ b/src/psu/PSUSensorMain.cpp
@@ -120,6 +120,8 @@ static constexpr auto sensorTypes = std::to_array<
     {"MAX20730", I2CDeviceType{"max20730", true}},
     {"MAX20734", I2CDeviceType{"max20734", true}},
     {"MAX20796", I2CDeviceType{"max20796", true}},
+    {"MAX20912", I2CDeviceType{"max20912", true}},
+    {"MAX20916", I2CDeviceType{"max20916", true}},
     {"MAX34451", I2CDeviceType{"max34451", true}},
     {"MP2856", I2CDeviceType{"mp2856", true}},
     {"MP2857", I2CDeviceType{"mp2857", true}},
-- 
2.39.3

